handlers/pipeline.go

```diff
@@ -183,12 +184,66 @@ func PipelineUpdate(c echo.Context) error {
 
 		// Update active pipelines
 		pipeline.GlobalActivePipelines.ReplaceByName(currentName, foundPipeline)
+	}
+
+	// Check if the periodic scheduling has been changed.
+	if !stringSliceEqual(foundPipeline.PeriodicSchedules, p.PeriodicSchedules) {
+		// We prevent side effects here and make sure
+		// that no scheduling is already running.
+		if foundPipeline.CronInst != nil {
+			foundPipeline.CronInst.Stop()
+		}
+		foundPipeline.CronInst = cron.New()
+
+		// Iterate over all cron schedules.
+		for _, cron := range p.PeriodicSchedules {
+			err := foundPipeline.CronInst.AddFunc(cron, func() {
+				_, err := schedulerService.SchedulePipeline(&foundPipeline, []gaia.Argument{})
+				if err != nil {
+					gaia.Cfg.Logger.Error("cannot schedule pipeline from periodic schedule", "error", err, "pipeline", foundPipeline)
```

Comment:
Why not return the error here?

Reply:
Because `AddFunc` only accepts functions without return values. This is executed in it's own go-routine.